strongbox-data-service/src/main/java/org/carlspring/strongbox/data/service/CommonCrudService.java

```diff
@@ -49,7 +59,7 @@ public void postConstruct()
     {
         identifyEntity(entity);
         
-        ReflectionUtils.doWithFields(entity.getClass(), (field) -> {
+        ReflectionUtils.doWithFields(this.getEntityClass(), (field) -> {
```

Comment:
I think we can try to drop this `getEntityClass()` method and use some utility which will expose target class from underlying `InvocationHandler`.